include/platform/mir/graphics/platform.h

```diff
@@ -396,7 +396,7 @@ class DmaBufDisplayAllocator : public DisplayAllocator
     {
     };
 
-    virtual auto framebuffer_for(std::shared_ptr<DmaBufBuffer> buffer)
+    virtual auto framebuffer_for(mir::graphics::DisplaySink& sink, std::shared_ptr<DMABufBuffer> buffer)
```

Comment:
You don't need this, as the `DisplayAllocator` is already bound to the `DisplaySink` (you get a `DisplayAllocator` *from* a `DisplaySink`, it's associated with that sink, and the lifetime is bound to that sink)